pkg/applier/applier.go

```diff
@@ -63,7 +63,51 @@ var (
 	maxRequestBytes = int64(1.5 * 1024 * 1024)
 )
 
-// applier is the default implimentation of the Applier interface.
+// Applier is a bulk client for applying a set of desired resource objects and
+// tracking them in a ResourceGroup inventory. This enables pruning objects
+// by removing them from the list of desired resource objects and re-applying.
+type Applier interface {
+	// Apply creates, updates, or prunes all managed resources, depending on
+	// the new desired resource objects.
+	// Returns the set of GVKs which were successfully applied and any errors.
+	// This is called by the reconciler when changes are detected in the
+	// source of truth (git, OCI, helm) and periodically.
+	Apply(ctx context.Context, desiredResources []client.Object) (map[schema.GroupVersionKind]struct{}, status.MultiError)
+	// Errors returns the errors encountered during apply.
+	// This method may be called while Destroy is running, to get the set of
+	// errors encounted so far.
+	Errors() status.MultiError
+}
+
+// Destroyer is a bulk client for deleting all the managed resource objects
+// tracked in a single ResourceGroup inventory.
+type Destroyer interface {
+	// Destroy deletes all managed resources.
+	// Returns any errors encountered while destorying.
+	// This is called by the reconciler finalizer when deletion propagation is
+	// enabled.
+	Destroy(ctx context.Context) status.MultiError
+	// Errors returns the errors encountered during destroy.
+	// This method may be called while Destroy is running, to get the set of
+	// errors encounted so far.
+	Errors() status.MultiError
+}
+
+// Dormammu is an incredibly powerful being that dwells in the Dark Dimension,
+// known as the Destroyer of Worlds. He can apply and destroy whole sets of
+// resource objects at will.
+//
+// Not to be confused with The Doctor, who was named the Destroyer of Worlds by
+// Davros, creator of the Daleks.
+//
+// Yes, the name is silly. But applier.ApplierDestroyer is repetitive and
+// doesn't pass lint checks.
+type Dormammu interface {
```

Comment:
Feel free to suggest a better name, if you like.
This one's cute and mildly confusing, but `applier.ApplierDestroyer` doesn't meet lint requirements due to repetition.

Reply:
Needs a `Bargain` method :)

Reply:
Dominator isn't bad. It does mean something in graph-theory. But how about Supervisor instead?

Reply:
Supervisor sounds good!

Reply:
Renamed to Supervisor.